### PR TITLE
Feature: Added faculty filtering to search bar (Search bar Redesign #818)

### PR DESCRIPTION
**Pull request by chanboba**
* added the ability to filter course by faculty 
* added a facultyName map to map the fetched faculty names to our stored names
* added deselecting functionality to the filter buttons
* ensured same area of study filtering only applies when no faculty is selected

image

image

### CODE REVIEW

File: client/src/components/controls/CourseSelect.tsx [hunk 1]
```diff
@@ -171,6 +195,13 @@ const CourseSelect: React.FC<CourseSelectProps> = ({ assignedColors, handleSelec
    */
   const getCourseArea = (courseCode: CourseCode) => courseCode.substring(0, 4);
 
+  /**
+   * @param faculty The faculty of the course
+   */
+  const handleFacultyClick = (faculty: string) => {
+    setSelectedFaculty(selectedFaculty == faculty ? "" : faculty);
```
Comment (Rayahhhmed) [inline client/src/components/controls/CourseSelect.tsx]: nit: ``===`` please

File: client/src/interfaces/Courses.ts [hunk 2]
```diff
@@ -20,4 +21,5 @@ export interface FetchedCourse {
   online: boolean;
   inPerson: boolean;
   career: string;
-}
+  faculty: string;
+}
```
Comment (Rayahhhmed) [inline client/src/interfaces/Courses.ts]: nit: add a new line

File: client/src/components/controls/CourseSelect.tsx [hunk 3]
```diff
@@ -403,4 +476,4 @@ const CourseSelect: React.FC<CourseSelectProps> = ({ assignedColors, handleSelec
   );
 };
 
-export default CourseSelect;
+export default CourseSelect;
```
Comment (Rayahhhmed) [inline client/src/components/controls/CourseSelect.tsx]: nit: add a new line

**Review by Rayahhhmed — CHANGES_REQUESTED** [review]

File: client/src/components/controls/CourseSelect.tsx [hunk 4]
```diff
@@ -136,10 +140,30 @@ const Career = styled('div')`
...
+  width: 100%;
+`;
+
+
 const CourseSelect: React.FC<CourseSelectProps> = ({ assignedColors, handleSelect, handleRemove }) => {
   const [options, setOptionsState] = useState<CoursesList>([]);
   const [inputValue, setInputValue] = useState<string>('');
   const [selectedValue, setSelectedValue] = useState<CoursesList>([]);
+  const [selectedFaculty, setSelectedFaculty] = useState<string>('');
+  const faculties = ["Art, Design & Architecture", "Law & Justice", "Medicine & Health", "Engineering", "Business School", "Science"];
```
Comment (Rayahhhmed) [inline client/src/components/controls/CourseSelect.tsx]: double ""

File: client/src/components/controls/CourseSelect.tsx [hunk 5]
```diff
@@ -274,6 +323,32 @@ const CourseSelect: React.FC<CourseSelectProps> = ({ assignedColors, handleSelec
...
+                  padding: '5px 10px',
+                  cursor: 'pointer',
+                  backgroundColor: selectedFaculty === faculty ? theme.palette.primary.main : theme.palette.secondary.light,
+                  color: selectedFaculty === faculty ? '#FFFFFF' : theme.palette.secondary.dark,
+                  textTransform: 'none',
+                  fontSize: '13px',
+                }}
+              >
+                {faculty}
+              </Button>
```
Comment (Shzmj) [inline client/src/components/controls/CourseSelect.tsx]: can we create a styled component for this to avoid inline styling plspsls

**Review by Shzmj — CHANGES_REQUESTED** [review]

**Review by Rayahhhmed — APPROVED** [review]
clgtm

**Review by Shzmj — APPROVED** [review]

**Pull request merged**